library/alloc/src/collections/btree/node.rs

```diff
@@ -47,8 +47,7 @@ const KV_IDX_CENTER: usize = B - 1;
 const EDGE_IDX_LEFT_OF_CENTER: usize = B - 1;
 const EDGE_IDX_RIGHT_OF_CENTER: usize = B;
 
-/// The underlying representation of leaf nodes.
-#[repr(C)]
```

Comment:
Why was this repr(C) removed? I'm pretty confident it is indeed still necessary for soundness of treating LeafNode as a prefix of InternalNode.

Reply:
Well, I'm pretty confident it is not. The introduction of the empty root node (#50352) made it necessary, but you simplified that away, so we don't need to constrain leaf layout. For a particular K and V, whatever layout the compiler chooses , the field in `InternalNode` has the same layout and the repr(C) on that guarantees that the internal node's alignment is at least as coarse as the leaf node's. The unit test `test_iter_mut_mutation` covers alignment of internal nodes. And from voluntary and involuntary experiments, I can tell the compiler is eager to reorder fields if you let your guard down.

Reply:
Yes, thinking more, that's correct -- if we want to take a prefix of LeafNode and treat that specially (e.g., assume parent is the first thing) then that would be different, but we don't do that.